include/sundials/priv/sundials_errors_impl.h

```diff
@@ -108,44 +173,62 @@ static inline void SUNHandleErrWithFmtMsg(int line, const char* func,
 #define SUNCheck(expr, code)
 #endif
 
-/* SUNCheckCallNoRet performs the SUNDIALS function call, and checks the
+/*
+   SUNCheckCallMsg performs the SUNDIALS function call, and checks the
    returned error code. If an error occured, then it will log the error, set the
-   last_err value, and call the error handler. */
+   last_err value, call the error handler, **and then return the error code**.
+
+   :param call: the function call
+   :param msg: an error message
+*/
 #if defined(SUNDIALS_ENABLE_ERROR_CHECKS)
-#define SUNCheckCallNoRetMsg(call, msg)                      \
+#define SUNCheckCallMsg(call, msg)                           \
   do {                                                       \
     SUNErrCode sun_chk_call_err_code_ = call;                \
     if (SUNHintFalse(sun_chk_call_err_code_ < 0))            \
     {                                                        \
       SUNHandleErrWithMsg(__LINE__, __func__, __FILE__, msg, \
                           sun_chk_call_err_code_, SUNCTX_);  \
-      (void)SUNContext_GetLastError(SUNCTX_);                \
+      return sun_chk_call_err_code_;                         \
     }                                                        \
   }                                                          \
   while (0)
 #else
-#define SUNCheckCallNoRetMsg(call, msg) (void)call
+#define SUNCheckCallMsg(call, msg) (void)call
 #endif
 
-/* Same as SUNCheckCallNoRet, but returns with the error code if an error
- * occured. */
+/*
+   SUNCheckCallNoRetMsg performs the SUNDIALS function call, and checks the
+   returned error code. If an error occured, then it will log the error, set the
+   last_err value, and call the error handler. **It does not return**.
+
+   :param call: the function call
+   :param msg: an error message
+*/
 #if defined(SUNDIALS_ENABLE_ERROR_CHECKS)
-#define SUNCheckCallMsg(call, msg)                           \
+#define SUNCheckCallNoRetMsg(call, msg)                      \
   do {                                                       \
     SUNErrCode sun_chk_call_err_code_ = call;                \
     if (SUNHintFalse(sun_chk_call_err_code_ < 0))            \
     {                                                        \
       SUNHandleErrWithMsg(__LINE__, __func__, __FILE__, msg, \
                           sun_chk_call_err_code_, SUNCTX_);  \
-      return sun_chk_call_err_code_;                         \
+      (void)SUNContext_GetLastError(SUNCTX_);                \
```

Comment:
It will, but that is what we want since the error is handled at this point.  If we did not clear the error, then the very next `SUNCheckLastErr` would try and handle the same error again.  Clearing the error fixes this problem and makes it behave similarly to `SUNCheckCall` around a function that returns an error code.

Reply:
Ok, I guess this is for internal functions where we don't currently return or check for an error. 

I think the MPI version of this function needs `SUNContext_GetLastError` as well.